### PR TITLE
Copy paste rois crop

### PR DESCRIPTION
**Pull request by will-moore**
This is the last major piece of ROIs functionality for the next release!
It allows copying and bulk editing of ROIs between panels without opening the full ROI viewer.
Also allows copying and pasting a crop region -> Rectangle and vice versa.

Functional testing is below, but any comment on workflow, usability etc would also be useful.

screen shot 2015-10-28 at 16 47 07

 - Copy & Paste ROIs
    - Draw some ROIs in the ROI dialog, OK to close.
    - The number of ROIs in selected panels should be shown at top of Labels tab.
    - Now you can copy all the ROIs on the panel with the Copy button.
    - Select another panel (or panels) on the figure and click Paste button (or keyboard shortcut)
    - Can also Delete all ROIs on selected panels with the Delete button
 - Bulk edit ROIs (color / width)
    - The Colour and Line width choosers on the ROI controls on the Labels tab should apply chosen colour or line width to all shapes in selected panels.

screen shot 2015-10-28 at 22 27 09

 - Copy & Paste crop region
    - In the Preview tab, zoom and pan the viewport, check that the View x, y, width height update.
    - Check that the "Crop" button works as before (launches crop dialog) and the "Reset" button reverts to uncropped panel.
    - Copy the crop region and paste onto another panel / panels
    - Select multiple panels with different view regions, should show '-' for x, y, width, height.

screen shot 2015-10-28 at 22 38 28

 - Copy ROIs Rectangle -> Crop region
    - Duplicate a panel by copying and pasting the panel
    - Copy a single Rectangle on the panel by selecting it and 'Copy' in the ROIs dialog.
    - Paste in the "View" controls to crop the other panel. Should show the same region highlighted by the rectangle in first panel.
 - Copy Crop region -> Rectangle
    - Again, copy and paste a panel to duplicate
    - Zoom and pan one panel to chosen view
    - Copy the view region, select the other panel and Paste this as a new Rectangle ROI.
    - Can then update the colour / line width of this as above

out0001

 - Crop dialog - pick regions from Clipboard or ROIs on the image
    - Copy some ROIs (including Rectangle(s)) to the clipboard
    - Open the Crop dialog (green button) on an image that has some Rectangular ROIs drawn on it
    - You should see that you can pick regions from the clipboard or from ROIs on the image and use these to crop the image
    - If you are cropping an image that has ROIs on it, you will be asked if you want to remove these ROIs when you crop it (since they will likely be outside or obscuring the crop region)

screen shot 2015-11-06 at 00 36 41

### CODE REVIEW

File: static/figure/css/figure.css [hunk 1]
```diff
@@ -495,7 +486,7 @@
     }
     .tab-pane .ui-slider-horizontal {
         height: 10px;
-        width: 150px;
+        width: 215 px;
```
Comment (ximenesuk) [inline static/figure/css/figure.css]: Trivial CSS style thing but you've introduced a space before the `px` and it's the only instance of a space before `px` in this file.

File: static/figure/js/views/right_panel_view.js [hunk 2]
```diff
@@ -78,14 +242,11 @@
         events: {
             "submit .new-label-form": "handle_new_label",
             "click .dropdown-menu a": "select_dropdown_option",
-            "click .edit_rois": "editRois",
-        },
-
-        editRois: function(event) {
-            $("#roiModal").modal("show");
+            // "click .edit_rois": "editRois",
```
Comment (ximenesuk) [inline static/figure/js/views/right_panel_view.js]: Is there a reason to leave in the commented out event given the function has been removed?

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [FIGURE-merge#3](https://ci.openmicroscopy.org/job/FIGURE-merge/3/). See the [console output](https://ci.openmicroscopy.org/job/FIGURE-merge/3/consoleText) for more details.
Possible conflicts:
  - PR #114 joshmoore 'Add support for Docker'
    - README.md

**Comment by will-moore** [comment]
@ximenesuk @bramalingam This got excluded today, but if you wanted to look at it, it's on the staging demo site at http://figure.staging.openmicroscopy.org/demo/#file/1

**Comment by ximenesuk** [comment]
Thanks @will-moore I'll take a look.

**Comment by ximenesuk** [comment]
This all seems to work as expected now that I have got my head around the workflow.

**Comment by ximenesuk** [comment]
Using a figure ROI to crop is working as expected. It'd be good to test more fully it with images containing ROIs against a server if possible.

**Comment by will-moore** [comment]
@ximenesuk Thanks - changes are on eel today. E.g. figure where image has OMERO ROIs is https://eel.openmicroscopy.org/merge/figure/file/12851 (user-3).

**Comment by ximenesuk** [comment]
@will-moore thanks, this one got overlooked today. I'll review it tomorrow along with https://github.com/ome/figure/pull/118

**Comment by ximenesuk** [comment]
Sorry @will-moore I did review this on Friday but seem to have failed to comment. I checked against eel using existing ROIs and the new workflows again. Everything looks okay. Good to merge.

**Pull request merged**